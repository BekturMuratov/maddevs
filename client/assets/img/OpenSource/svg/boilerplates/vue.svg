<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>