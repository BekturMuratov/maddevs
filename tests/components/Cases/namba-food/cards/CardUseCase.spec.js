import CardUseCase from '@/components/Cases/nambafood/cards/CardUseCase.vue'
import { render, screen } from '@testing-library/vue'

const props = {
  title: 'some-title',
  description: 'some-description',
  pictureFolder: '',
  fileName: 'some-filename',
  fileExtension: 'png',
  alt: 'alt',
}

const stubs = ['Picture', 'ClientOnly']

describe('CardUseCase component', () => {
  it('should render correctly', () => {
    const { container } = render(CardUseCase, {
      props,
      stubs,
      mocks: {
        $getMediaFromS3: () => 'img.jpg',
      },
    })

    expect(container).toMatchSnapshot()
  })

  it('should correctly render classname', () => {
    const classList = {
      uniqueСlass: 'uniqueСlass',
    }
    const { html } = render(CardUseCase, {
      props: {
        ...props,
        classList,
      },
      stubs,
      mocks: {
        $getMediaFromS3: () => 'img.jpg',
      },
    })

    expect(screen.getByText(props.title)).toBeTruthy()
    expect(html()).toContain(`card-use-case__${classList.uniqueСlass}-title`)
  })
})
