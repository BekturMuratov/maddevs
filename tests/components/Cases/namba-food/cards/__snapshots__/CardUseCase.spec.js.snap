// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardUseCase component should render correctly 1`] = `
<div>
  <div
    class="card-use-case"
  >
    <clientonly-stub>
      <h3
        class="card-use-case__title case_title_h4 card-use-case__undefined-title"
      >
        some-title
      </h3>
    </clientonly-stub>
     
    <div
      class="card-use-case__description-wrapper card-use-case__undefined-description-wrapper"
    >
      <p
        class="case_paragraph card-use-case__paragraph card-use-case__undefined-paragraph"
      >
        
      some-description
    
      </p>
       
      <div
        class="card-use-case__desktop-picture card-use-case__undefined-picture"
      >
        <picture-stub
          alt="alt"
          extension="png"
          file="some-filename"
          folder=""
          height="0"
          width="0"
        />
      </div>
       
      <div
        class="card-use-case__mobile-picture card-use-case__undefined-mobile-picture"
      >
        <picture-stub
          alt="alt"
          extension="png"
          file="some-filename-mobile"
          folder=""
          height="0"
          width="0"
        />
      </div>
    </div>
  </div>
</div>
`;
